proposals/gc/MVP.md

```diff
@@ -75,15 +74,24 @@ New abbreviations are introduced for reference types in binary and text format,
 * `i31ref` is a new reference type
   - `i31ref == (ref i31)`
 
-* `rtt <n>? <typeidx>` is a new reference type
-  - `(rtt <n>? $t) == (ref (rtt <n>? $t))`
+* `rtt <typeidx>` is a new reference type
+  - `(rtt $t) == (ref (rtt $t))`
 
 
 #### Type Definitions
 
-* `deftype` is a new category of types that generalises the existing type definitions in the type section
-  - `deftype ::= <functype> | <structtype> | <arraytype>`
+* `deftype` is the syntax for an entry in the type section, generalising the existing syntax
+  - `deftype ::= <subtype> | rec <subtype>*`
   - `module ::= {..., types vec(<deftype>)}`
+  - a `rec` definition defines a group of mutually recursive types that can refer to each other; it thereby defines several type indices at a time
```

Comment:
Do we intend to support groups of size 0? It would slightly simplify the Binaryen implementation if we did not.

Reply:
Ah, good question. Preferably, I'd go with what we do everywhere else in Wasm, which is avoiding special rules for edge cases. How much of a simplification would it be to rule it out?

Reply:
That's fine. It will mean adding the ability to shrink a `TypeBuilder` in Binaryen, but that's not too complicated.